### PR TITLE
DVCC Max charge current setting modal layout issue

### LINKED ISSUE #1271: DVCC Max charge current setting modal layout issue
User report:

IMG_2067

on my phone, portrait:

IMG_2065

on my phone, landscape:

IMG_2066

### PR DESCRIPTION
**Pull request by DanielMcInnes**
Fixes #1271.

Tested on linux desktop firefox & chrome, android browser.

@blammit , could you please test on iphone?
@chriadam , could you please test on windows?

### CODE REVIEW

**Comment by chriadam** [comment]
I assume it's because DialogLayer.qml has `anchors.fill: parent` and that size will be affected by scaling, so when it constructs the dialog it thinks its available size is smaller.  But not sure.

**Comment by DanielMcInnes** [comment]
I added some debug console outs on the scaling in Main.qml, and the Max Charge Current dialog scales correctly with the rest of the app down to 60%, and then scales down at a slower rate than the rest of the app. No idea why.

**Review by chriadam — APPROVED** [review]
Nice find.  I've tested windows native builds, and wasm (chrome on windows) also, and the dialogs now scale nicely.  LGTM.

**Review by blammit — APPROVED** [review]

File: components/dialogs/ModalDialog.qml [hunk 1]
```diff
@@ -37,8 +37,15 @@ T.Dialog {
...
 
 	anchors.centerIn: parent
-	implicitWidth: Theme.geometry_modalDialog_width
-	implicitHeight: Theme.geometry_modalDialog_height
+
+	/*
+	If you specify implicitWidth & implicitHeight here, and shrink the browser (or desktop app) window from 100% -> 0%,
+	the dialog scales correctly with the rest of the app down to ~60%, and then scales down at a faster rate than the
+	rest of the app. Specifying 'width' and 'height' instead of 'implicitWidth' and 'implicitHeight' for ModalDialog
+	makes it scale properly when you shrink the window. See https://bugreports.qt.io/browse/QTBUG-127068
```
Comment (blammit) [inline components/dialogs/ModalDialog.qml:45]: Nice, glad you filed the Qt bug as well.

**Pull request merged**